src/state_machine/mod.rs

```diff
@@ -45,9 +45,9 @@ pub enum DkgError {
     /// DKG end timeout
     #[error("DKG end timeout, waiting for {0:?}")]
     DkgEndTimeout(Vec<u32>),
-    /// DKG end failure
+    /// DKG end failure, with a map of signer_id to reported errors and new malicious signer_ids
     #[error("DKG end failure")]
-    DkgEndFailure(HashMap<u32, DkgFailure>),
+    DkgEndFailure(HashMap<u32, DkgFailure>, HashSet<u32>),
```

Comment:
So before if a signer thought that there was a failure because another signer was being malicious, they didn't report who they thought was being malicious?

Reply:
Signers always reported who they thought was malicious.  But the coordinator was not reporting who they thought was malicious to the caller; the coordinator just added them to their internal malicious maps.